openml-h2o/src/main/java/com/feedzai/openml/h2o/params/ParametersBuilderUtil.java

```diff
@@ -128,9 +135,75 @@ private ParametersBuilderUtil() { }
                 .filter(modelParam -> !"calibrate_model".equals(modelParam.getName()))
                 // We already pass a seed always whenever possible.
                 .filter(modelParam -> !"seed".equals(modelParam.getName()))
+                // We only want the fields that are present in the list of fields of the algorithm class.
+                .filter(modelParam -> usefulFields.contains(modelParam.getName()))
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Returns the fields that are set in the given {@linkplain ModelParametersSchemaV3 algorithmClass} as useful for
+     * the algorithm.
+     *
+     * @param algorithmClass The algorithm class to get the useful fields.
+     * @param allParameters  All the parameters to be added if useful fields is not provided by the class.
+     * @return The {@link List} with the useful fields for the given algorithm class.
+     * @since @@@feedzai.next.release@@@
+     */
+    private static List<String> getUsefulFields(final Class<? extends ModelParametersSchemaV3> algorithmClass,
+                                                final Set<String> allParameters) {
+
+        final Field[] declaredFields = algorithmClass.getDeclaredFields();
+        final List<Field> staticFields = new ArrayList<>();
+
+        for (final Field field : declaredFields) {
+            if (isStatic(field.getModifiers())) {
+                staticFields.add(field);
+            }
+        }
+
+        final Optional<Field> usefulFieldsArray = staticFields.stream()
+                .filter(fields -> "fields".equalsIgnoreCase(fields.getName()))
```

Comment:
Is there a reason to build "staticFields" first? Can't we run this filter directly over "declaredFields"?

Reply:
After looking at the code I have no idea why I've done that. Fixed :+1: